lib/xrandr.rb

```diff
@@ -154,10 +164,37 @@ class Mode
     attr_reader :resolution, :rate, :current, :preferred
 
     def initialize(args={})
-      @resolution = args.fetch :resolution
+      @resolution = Resolution.new(args.fetch :resolution)
       @rate = args.fetch :rate
       @current = args.fetch :current
       @preferred = args.fetch :preferred
     end
   end
+
+  ##
+  # This class converts a resolution string of the form "1234x5678" (arbitrary
+  # digit count) into a form with discrete numeric x and y components.
+  # It implements the comparable interface on the basis of total pixel count.
+  class Resolution
+    include Comparable
+    MATCHER = /(?<x>\d+)x(?<y>\d+)/
+    attr_accessor :x
+    attr_accessor :y
```

Comment:
I would rather have this 2 as `attr_reader` and make it resolution "less mutable"
Also, I believe the names we use for resolution attributes are :width and :height 
Finally, the CODE STYLE COP :policeman:  says he prefers attributes in a single line: `attr_reader :width, :height`